src/main/msc/usbd_storage.h

```diff
@@ -29,18 +29,24 @@
 
 #ifdef USE_HAL_DRIVER
 extern USBD_StorageTypeDef *USBD_STORAGE_fops;
-#ifdef USE_SDCARD
+#ifdef USE_SDCARD_SDIO
 extern USBD_StorageTypeDef USBD_MSC_MICRO_SDIO_fops;
 #endif
+#ifdef USE_SDCARD_SPI
+extern USBD_StorageTypeDef USBD_MSC_MICRO_SD_SPI_fops;
+#endif
 #ifdef USE_FLASHFS
 extern USBD_StorageTypeDef USBD_MSC_EMFAT_fops;
 #endif
-#else
+#else // USE_HAL_DRIVER
 extern USBD_STORAGE_cb_TypeDef *USBD_STORAGE_fops;
-#ifdef USE_SDCARD
+#ifdef USE_SDCARD_SDIO
 extern USBD_STORAGE_cb_TypeDef USBD_MSC_MICRO_SDIO_fops;
 #endif
+#ifdef USE_SDCARD_SPI
+extern USBD_STORAGE_cb_TypeDef USBD_MSC_MICRO_SD_SPI_fops;
+#endif
 #ifdef USE_FLASHFS
 extern USBD_STORAGE_cb_TypeDef USBD_MSC_EMFAT_fops;
 #endif
-#endif
+#endif // USE_HAL_DRIVER
```

Comment:
I realize this section isn't all new, but won't this break if a target has both SD card and on-board flash (and the user selected the SD card for logging).

Reply:
@etracer65 
`usbd_storage.h` only defines function vector table for configured sources (sdcard on spi, sdcard on sdio and flashfs).  Are you talking about some logic in `drivers/usb_msc_f{4,7}xx.c` where actual switching is done?

Reply:
Never mind - I misunderstood what was going on.